src/simulation/m_bubbles.fpp

```diff
@@ -398,40 +450,57 @@ contains
         !!  @param fpb Current internal bubble pressure
         !!  @param fmass_v Current mass of vapour
         !!  @param iR0 Bubble size index (EE) or bubble identifier (EL)
+        !!  @param Tw_l Physical bubble parameters (pass in from global parameters)
+        !!  @param R_v_l Physical bubble parameters (pass in from global parameters)
+        !!  @param pv_l Physical bubble parameters (pass in from global parameters)
+        !!  @param pb0_l Physical bubble parameters (pass in from global parameters)
+        !!  @param mass_n0_l Physical bubble parameters (pass in from global parameters)
+        !!  @param mass_v0_l Physical bubble parameters (pass in from global parameters)
+        !!  @param Pe_T_l Physical bubble parameters (pass in from global parameters)
+        !!  @param Re_trans_T_l Physical bubble parameters (pass in from global parameters)
+        !!  @param k_mw_l Bubble Wall properties (pass in from bubbles module)
+        !!  @param thermal_l Thermal Behavior (pass in from global parameters)
+        !!  @param bubbles_lagrange_l Lagrangian subgrid bubble model switch (pass in from global parameters)
         !!  @param fbeta_t Mass transfer coefficient (EL)
         !!  @param fR_m Mixture gas constant (EL)
         !!  @param fgamma_m Mixture gamma (EL)
-    function f_bpres_dot(fvflux, fR, fV, fpb, fmass_v, iR0, fbeta_t, fR_m, fgamma_m)
+    pure function f_bpres_dot(fvflux, fR, fV, fpb, fmass_v, iR0, Tw_l, R_v_l, pv_l, pb0_l, mass_n0_l, mass_v0_l, Pe_T_l, Re_trans_T_l, k_mw_l, thermal_l, bubbles_lagrange_l, fbeta_t, fR_m, fgamma_m)
         !$acc routine seq
         real(wp), intent(in) :: fvflux
         real(wp), intent(in) :: fR
         real(wp), intent(in) :: fV
         real(wp), intent(in) :: fpb
         real(wp), intent(in) :: fmass_v
         integer, intent(in) :: iR0
+        real(wp), intent(in) :: Tw_l, R_v_l, pv_l
+        real(wp), dimension(:), allocatable, intent(in) :: pb0_l, mass_n0_l, mass_v0_l, Pe_T_l
+        real(wp), dimension(:), allocatable, intent(in) :: Re_trans_T_l
+        real(wp), intent(in) :: k_mw_l
+        integer, intent(in) :: thermal_l
+        logical, intent(in) :: bubbles_lagrange_l
```

Comment:
Similar to above, since these are `intent(in)` why was the refactor necessary to make it `pure`?